File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;
class Solution {
    public int[] solution(int k, int[] score) {
        if(k>score.length){
            int[] answer = new int[score.length];
            int min =2000;
            for(int i=0; i < score.length;i++){
                min = Math.min(min,score[i]);
                answer[i] = min;
            }
        return answer;
        }
       PriorityQueue<Integer> hallOfFame = new PriorityQueue<>();
        int[] ans = new int[score.length];
        for(int i =0; i < k; i++){
            hallOfFame.add(score[i]);
            ans[i] = hallOfFame.peek();
        }
        for(int i =k; i < score.length;i++){
            if(score[i]>=hallOfFame.peek()) {
                hallOfFame.add(score[i]); hallOfFame.poll();}
            ans[i] = hallOfFame.peek();
        }
        return ans;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    hall_of_fame = score[:k]
    range_ = range(k) if k < len(score) else range(len(score))
    answer = [min(hall_of_fame[:i+1]) for i in range_]
    
    if k > len(score):
        return answer
    
    for i in range(k,len(score)):
        if score[i] > min(hall_of_fame):
            hall_of_fame.remove(min(hall_of_fame))
            hall_of_fame.append(score[i])
        answer.append(min(hall_of_fame))
    return answer